test/utils/experiment-provider.test.js

```diff
@@ -0,0 +1,140 @@
+/* eslint-disable no-underscore-dangle */
+import { expect } from '@esm-bundle/chai';
+import getExperimentData from '../../unitylibs/utils/experiment-provider.js';
+
+describe('getExperimentData', () => {
+  beforeEach(() => {
+    // Mock window._satellite
+    window._satellite = { track: () => {} };
+  });
+
+  afterEach(() => {
+    delete window._satellite;
+  });
+
+  it('should reject when target fetch fails', async () => {
+    window._satellite.track = (event, options) => {
+      setTimeout(() => {
+        if (typeof options.done === 'function') options.done(null, new Error('Test error'));
+      }, 0);
+    };
```

Comment:
I see variations of this "duplicated" in other parts of this file. Could we somehow abstract it out into a separate function to remove some redundancies?